packages/assetpack/src/image/utils/compressSharp.ts

```diff
@@ -1,12 +1,12 @@
 import type { AvifOptions, JpegOptions, PngOptions, WebpOptions } from 'sharp';
-import type { CompressImageData, CompressOptions } from '../compress.js';
+import type { CompressImageData, CompressImageDataResult, CompressOptions } from '../compress.js';
 
 export async function compressSharp(
     image: CompressImageData,
     options: CompressOptions
-): Promise<CompressImageData[]>
+): Promise<CompressImageDataResult[]>
 {
-    const compressed: CompressImageData[] = [];
+    const compressed: CompressImageDataResult[] = [];
```

Comment:
instead of awaiting all of the buffers one by one what about doing this at the end

```ts
export async function compressSharp(
    image: CompressImageData,
    options: CompressOptions
): Promise<CompressImageDataResult[]>
{
    const compressed: CompressImageData[] = [];
    const sharpImage = image.sharpImage;

    if (image.format === '.png' && options.png)
    {
        compressed.push({
            format: '.png',
            resolution: image.resolution,
            sharpImage: sharpImage.clone().png({ ...options.png as PngOptions, force: true }),
        });
    }

    ...

    const results = await Promise.all(compressed.map(async (result) => ({
        ...result,
        buffer: await result.sharpImage.toBuffer()
    })));

    return results;
}
```